### ISSUE TITLE
linter: Insufficient memory to create fixed-size allocator pool

### ISSUE THREAD

**Issue by thobas-dnvgl** [oxc-project/oxc#20063]
### What version of Oxlint are you using?

1.51.0

### What command did you run?

`oxlint --report-unused-disable-directives`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint';

export default defineConfig({
	categories: {
		correctness: 'error',
	},
	plugins: ['eslint', 'import', 'react-perf', 'react', 'typescript', 'unicorn', 'vitest'],
	jsPlugins: ['@eslint-react/eslint-plugin'],
	options: {
		typeAware: true,
		typeCheck: true,
	},
	rules: {
		'import/no-cycle': 'error',
		'no-irregular-whitespace': 'off',
		'no-unused-vars': 'off',
		'react/exhaustive-deps': 'off',
		'react/no-children-prop': 'off',
		'react/only-export-components': 'error',
		'typescript/no-floating-promises': 'off',
		'typescript/unbound-method': 'off',
	},
	overrides: [
		{
			jsPlugins: ['eslint-plugin-storybook'],
			files: ['**/*.stories.tsx'],
			rules: {
				'typescript/no-floating-promises': 'off',
			},
		},
	],
	ignorePatterns: ['src/tsr/routeTree.gen.ts'],
});
```

### What happened?

Note that I set `RUST_BACKTRACE` to `full`.

```
> oxlint --report-unused-disable-directives


thread 'tokio-runtime-worker' (15148) panicked at crates\oxc_allocator\src\pool\fixed_size.rs:170:18:
Insufficient memory to create fixed-size allocator pool
stack backtrace:
   0:     0x7ff823e10cd2 - napi_register_module_v1
   1:     0x7ff823e1cb01 - napi_register_module_v1
   2:     0x7ff823df2204 - napi_register_module_v1
   3:     0x7ff823e069c0 - napi_register_module_v1
   4:     0x7ff823e067d8 - napi_register_module_v1
   5:     0x7ff823e06cf2 - napi_register_module_v1
   6:     0x7ff823e06ac9 - napi_register_module_v1
   7:     0x7ff823e0196f - napi_register_module_v1
   8:     0x7ff823def12e - napi_register_module_v1
   9:     0x7ff823e9c2fd - napi_register_module_v1
  10:     0x7ff823dcf5bd - napi_register_module_v1
  11:     0x7ff823951611 - napi_register_module_v1
  12:     0x7ff82386d30c - napi_register_module_v1
  13:     0x7ff8236d9fa2 - <unknown>
  14:     0x7ff823693873 - <unknown>
  15:     0x7ff8236a6878 - <unknown>
  16:     0x7ff8237dff3b - napi_register_module_v1
  17:     0x7ff8237de16b - napi_register_module_v1
  18:     0x7ff8237d7850 - napi_register_module_v1
  19:     0x7ff8237d42ed - napi_register_module_v1
  20:     0x7ff8237d4b21 - napi_register_module_v1
  21:     0x7ff823dfaa18 - napi_register_module_v1
  22:     0x7ff902f1e8d7 - BaseThreadInitThunk
  23:     0x7ff904f0c40c - RtlUserThreadStart
```

**Comment by camc314** [comment]
This sounds like https://github.com/oxc-project/oxc/issues/19395 cc @overlookmotel 

Can you share:

 - some details of the machine you're running this on?
 - how much RAM
 - how much free disk space?


Thanks

**Comment by overlookmotel** [comment]
Yes, this is a duplicate of #19395. Closing in favour of that issue. But @thobas-dnvgl, please could you share the details of your machine on that issue? (as per previous comment)

**Comment by thobas-dnvgl** [comment]
- This is a (work provided) Dell precision 3591.
- 32Gb RAM
- I am using 372 GB of disk space out of 954GB

I cannot reproduce the crash though, the laptop was probably under heavy load.
